main.cpp

```diff
@@ -121,385 +163,563 @@ struct CarWash
+    int sampSize(); // used in UDT 10
+    void playForward(); // used in UDT 10
 };
 
-/*
-Thing 6) Global settings
-5 properties:
-    1) Gain level in dB
-    2) Pan setting (-50 to 50)
-    3) Number of voices
-    4) Sampler playback mode
-    5) Transposition in semitones 
-3 things it can do:
-    1) change amplitude of sample
-    2) pan audio in a stereo image
-    3) transpose audio in terms of musical intervals
- */
+void WaveformDisplay::transientPositions(bool bpmOn, float quarterSubdivision, int bpmValue = 120)
+{
+    if(bpmOn == true)
+    {
+        quarterSubdivision = 60 / bpmValue;
+    }
+    bpm += bpmValue;
+}
+
+void WaveformDisplay::playbackHeadPos(float samplePos)
+{
+    samplePos = 0.0f;
+}
+
+void WaveformDisplay::loopPoints(float startPos, float endPos, bool isLoopFunctionOn)
+{
+    if(isLoopFunctionOn == true)
+    {
+        loopHighlightColour = "Orange";
+    }
+    else
+    {
+        loopHighlightColour = "black";
+    }
+    startPos = 1.1f;
+    endPos = 2.3f;
+}
+
+int WaveformDisplay::sampSize()
+{
+    return 1;
+}
+
+void WaveformDisplay::playForward()
+{
+
+}
+// UDT 6 // complete
+
 struct GlobalSettings
 {
-    //Gain level in dB
     double gainLevel = 0.0;
-    //Pan setting (-50 to 50)
     int panPosition = -23;
-    //Number of voices
     int numVoices = 3;
-    //Sampler playback mode
     std::string playbackMode = "Classic";
-    //Transposition in semitones 
-    int transpose = -4;
-
-    //change amplitude of sample
-    double changeAmp(double gainLevel, double maxGain = 6); 
-    //pan audio in a stereo image
-    int panAudio(int panPosition, bool sampleIsStereo = false);
-    //transpose audio in terms of musical intervals
-    int transposeAudio(int transpose, int highestValue = 24, int lowestValue = -24);
+    int transpose = 0;
+
+    void changeAmp(double gainLevelChange); 
+    void panAudio(bool sampleIsStereo, bool outputStereo, int changePanPosition);
+    void transposeAudio(int changePitch);
 };
 
-/*
-Thing 7) Filter
-5 properties:
-    1) Frequency cutoff in Hz
-    2) Resonance as a percentage
-    3) Filter type selection
-    4) Distortion amount in dB
-    5) Key tracker
-3 things it can do:
-    1) affect frequency content of audio
-    2) distort frequency content of audio
-    3) adjust frequency cutoff of filter based on user MIDI note value input
- */
-struct Filter
+void GlobalSettings::changeAmp(double gainLevelChange = 2)
+{
+    gainLevel += gainLevelChange;
+}
+
+void GlobalSettings::panAudio(bool sampleIsStereo, bool outputStereo, int changePanPosition = 0)
+{
+    if(sampleIsStereo == true)
+    {
+        outputStereo = true;
+    }
+    else
+    {
+        outputStereo = false;
+    }
+    changePanPosition = 0;
+    panPosition += changePanPosition;
+}
+
+void GlobalSettings::transposeAudio(int changePitch = 2)
+{
+    if(changePitch > 0)
+    {
+        transpose += changePitch;
+    }
+    else 
+    {
+        transpose -= changePitch;
+    }
+}
```

Comment:
```
{
   transpose += changePitch;
}
```
if `changePitch` is negative, it'll be subtracted from `transpose`.  no need for the `if/else`
- [ ] revise